doc/command_line/field_processing.md

```diff
@@ -0,0 +1,124 @@
+# Field Processing
+
+Ruby supports <i>field processing</i>.
+
+This means that when certain command-line options are given,
+the invoked Ruby program can process input line-by-line.
+
+## About the Examples
+
+Examples here assume that this code (creating file `desiderata.txt`)
+has been executed:
+
+```
+text = <<EOT
+Go placidly amid the noise and the haste,
+and remember what peace there may be in silence.
+As far as possible, without surrender,
+be on good terms with all persons.
+EOT
+File.write('desiderata.txt', text)
+```
+
+The examples also use command-line option `-e`,
+which passes the Ruby code to be executed on the command line itself:
+
+```sh
+$ ruby -e "puts 'Hello, World.'"
+```
+
+## Option `-n`
+
+Option `-n` runs your program in a Kernel#gets loop:
+
+```
+while gets
+  # Your Ruby code.
+end
+```
+
+Note that `gets` reads the next line and sets global variable `$_`
+to the last read line;
+note also that character `'$'` must be escaped as `'\$'`:
```

Comment:
That escape is needed when enclosed in double quotes or no quotes.
Is it for the sake of Windows, not to use single quotes?

Reply:
Fixed (7 places).